src/config/ConfigHelper.ts

```diff
@@ -315,13 +330,8 @@ export class ConfigHelper {
         addresses
       )
     }
-    config.confidentialEVM =
```

Comment:
since you removed the check here... what about  all the addresses/chains coming from this line?:
`let contractAddressesConfig = this.getAddressesFromEnv(config.network, addresses)`
which value will these have for `sdk` field?  I think **None** :-)
only the hardcoded ones on `configHelperNetworks` will have it

Reply:
I updated the config object within `getConfig`, contractAddressesConfig contains only contract addresses, so I set the sdk instead inside config object, not the addresses. Let me know if I missed anything from your suggestion.

> the main issue is not addressed... this.config will be null everywhere, as we are no passing the Config to any of the classes, so i don't see how this line can ever evaluate to 'true'

For the config, it shouldn't be null, now since SmartContract class handles the creation in the constructor.